### PR TITLE
IO_toposort

### PR DESCRIPTION
**Pull request by ReyhaneAskari**
This fixes a minor error generated in docs for #5042 .

### CODE REVIEW

File: theano/gof/opt.py [hunk 1]
```diff
@@ -679,7 +681,7 @@ def process_node(self, fgraph, node):
                                in zip(node_inputs_assert_removed,
                                       cand_inputs_assert_removed))
 
-            if inputs_match and node.op == candidate.op:
+            if inputs_match:
```
Comment (nouiz) [inline theano/gof/opt.py]: Revert both changes in this file. This should be a separate PR that will need its own timing.

**Comment by nouiz** [comment]
The PR contain a merge commit. Can you do a rebase to get rid of it? We prefer to not have merge commits in PR.

**Comment by ReyhaneAskari** [comment]
Apparently there is a problem with the version of gpuarray that is used by Jenkins. We can get back to this after the fix.

**Comment by ReyhaneAskari** [comment]
jenkins test this

**Comment by nouiz** [comment]
jenkins test this

**Comment by nouiz** [comment]
Can you rebase this PR on the current master?
